src/opm/parser/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1348,8 +1352,30 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
 
     std::vector<double> EclipseGrid::activeVolume() const {
         std::vector<double> active_volume( this->m_nactive );
-
-        #pragma omp parallel for schedule(static)
+#ifdef  HAVE_OPENMP
+/*
+  Using omp_get_num_procs() to determine the number of threads here, instead of
+  relying on the default omp_get_max_threads() is quite ugly. The rational goes
+  as follows:
+
+  1. This function is short compared to typical simulation time of a full
+     opm/flow simulation, but it is important for interactive performance - i.e.
+     getting to the point where the simulator actually starts simulating.
+
+  2. For a parallel MPI based run this particular function will typically be run
+     by the process which is parsing the input file, while the other processes
+     will be idle.
+
+  3. If OMP_NUM_THREADS / omp_set_num_threads() has been used to limit the
+     number of threads used by OpenMP that is typically inspired by the
+     simulation phase where a much lower number is used - current default is 2.
+     Since the setting of num_threads is typically done to cater for a different
+     situation - i.e. the assemble of equations it is not really relevant to
+     this particular functiona and we use omp_get_num_procs() here.
```

Comment:
OMP_NUM_THREADS is an absolute authority, you cannot use more (size of thread pool). Thus this only works if the application has previously artificially limited the maximum number of threads, but user has not imposed a maximum using the env var. So this needs some rewording.

Reply:
OK; I must admit I don't really understand how it should phrased - I am &epsilon; away from just removing everything related to thread manipulations.

Reply:
remove the OMP_NUM_THREADS in front, and add a comment. 'If user has specified OMP_NUM_THREADS this still applies'.